app/client.jsx

```diff
@@ -14,11 +15,25 @@ const store = configureStore(initialState, browserHistory);
 const history = syncHistoryWithStore(browserHistory, store);
 const routes = createRoutes(store);
 
+/**
+ * Callback function handling frontend route changes.
+ */
+function onUpdate() {
+  // Prevent duplicate fetches when first loaded.
+  if (window.__INITIAL_STATE__ !== null) {
+    window.__INITIAL_STATE__ = null;
```

Comment:
on second thought, why do we have to reset the `__INITIAL_STATE__` to null if we're returning? Not sure if this line is needed?